packages/kit/src/internal/esm.ts

```diff
@@ -4,16 +4,27 @@ import { createJiti } from 'jiti'
 import { captureStackTrace } from 'errx'
 
 export interface ResolveModuleOptions {
+  /** @deprecated use `url` with URLs pointing at a file - never a directory */
   paths?: string | string[]
+  url?: URL | URL[]
 }
 
+export function directoryToParentURL (directory: string) {
+  return new URL('./index.js', pathToFileURL(directory.replace(/\/?$/, '/')))
```

Comment:
I think this could be simplified. We don't need a `URL` instance but simple string starting with `file://`.

If this util can also trust input is a dir path, we can use `filePathToURL(join(dir, '_index.js'))` (which returns a string).